tests/ruling/utils.ts

```diff
@@ -0,0 +1,194 @@
+import * as fs from "fs";
```

Comment:
This is much more than a utils file. It contains the actual ruling logic. I think we should rename it ruling.ts and if there are some things that we think do not belong to ruling.ts we can put them in a new utils file. Maybe just what is not called from `runRules`.